torchvision/prototype/models/depth/stereo/crestereo.py

```diff
@@ -0,0 +1,1020 @@
+    num_cross_attention_layers: int,
+    self_attention_type: str,
+    cross_attention_type: str,
+    **kwargs,
+) -> CREStereo:
+
+    feature_encoder = kwargs.pop("feature_encoder", None) or raft.FeatureEncoder(
+        block=feature_encoder_block,
+        layers=feature_encoder_layers,
+        strides=feature_encoder_strides,
+        norm_layer=feature_encoder_norm_layer,
+    )
+
+    if feature_encoder.output_dim % corr_groups != 0:
+        raise ValueError(
+            f"Final ``feature_encoder_layers`` size should be divisible by ``corr_groups`` argument."
+            f"Feature encoder output size : {feature_encoder.output_dim}, Correlation groups: {corr_groups}."
+        )
+
+    motion_encoder = kwargs.pop("motion_encoder", None) or raft.MotionEncoder(
+        in_channels_corr=corr_groups * int(np.prod(corr_search_window_1d)),
+        corr_layers=motion_encoder_corr_layers,
+        flow_layers=motion_encoder_flow_layers,
+        out_channels=motion_encoder_out_channels,
+    )
+
+    out_channels_context = feature_encoder_layers[-1] - recurrent_block_hidden_state_size
+    recurrent_block = kwargs.pop("recurrent_block", None) or raft.RecurrentBlock(
+        input_size=motion_encoder.out_channels + out_channels_context,
+        hidden_size=recurrent_block_hidden_state_size,
+        kernel_size=recurrent_block_kernel_size,
+        padding=recurrent_block_padding,
+    )
+
+    flow_head = kwargs.pop("flow_head", None) or raft.FlowHead(
+        in_channels=out_channels_context, hidden_size=flow_head_hidden_size
+    )
+
+    update_block = raft.UpdateBlock(motion_encoder=motion_encoder, recurrent_block=recurrent_block, flow_head=flow_head)
+
+    self_attn_block = LocalFeatureTransformer(
+        dim_model=feature_encoder.output_dim,
+        num_heads=num_attention_heads,
+        attention_directions=["self"] * num_self_attention_layers,
+        attention_type=self_attention_type,
+    )
+
+    cross_attn_block = LocalFeatureTransformer(
+        dim_model=feature_encoder.output_dim,
+        num_heads=num_attention_heads,
+        attention_directions=["cross"] * num_cross_attention_layers,
+        attention_type=cross_attention_type,
+    )
+
+    model = CREStereo(
+        feature_encoder=feature_encoder,
+        update_block=update_block,
+        flow_head=flow_head,
+        self_attn_block=self_attn_block,
+        cross_attn_block=cross_attn_block,
+        feature_downsample_rates=feature_downsample_rates,
+        correlation_groups=corr_groups,
+        search_window_1d=corr_search_window_1d,
+        search_window_2d=corr_search_window_2d,
+        search_dilate_1d=corr_search_dilate_1d,
+        search_dilate_2d=corr_search_dilate_2d,
+    )
+
+    if weights is not None:
+        model.load_state_dict(weights.get_state_dict(progress=progress))
+
+    return model
+
+
+class CREStereo_Weights(WeightsEnum):
+    pass
+
+
+class CREStereo_B_Weights(CREStereo_Weights):
```

Comment:
We usually just inherit from `WeightsEnum`.
Another thing, do you expect to have variant other than `crestereo_b` ? If not, I think we can just name it `crestereo`.

Reply:
I mostly expect to have variants that are pretrained at either different resolutions, different number of optimization steps, different number of datasets or different losses.

Reply:
In that case, we can have `CREStereo_B_Weights (WeightsEnum)` and you can remove the class `CREStereo_Weights`